apps/arweave/src/ar_downloader.erl

```diff
@@ -48,18 +48,29 @@ fill_to_capacity(Peers, ToWrite, BI) ->
 		unavailable ->
 			%% Block is not found locally, download it and its transactions.
 			fill_to_capacity2(Peers, RandHash, ToWrite, NewBI);
-		B ->
-			case [TX || TX <- ar_storage:read_tx(B#block.txs), not is_record(TX, tx)] of
-				[] ->
-					%% All of the TXs are found locally. Continue...
-					fill_to_capacity(
-						Peers,
-						lists:delete(RandHash, ToWrite),
-						NewBI
-					);
-				_ ->
+		B = #block{txs = TXS0} ->
+			{TXS, TXIDs} =
```

Comment:
`ar_storage:read_block` always returns a block without transactions, just transaction identifiers. The `[TX || TX <- ar_storage:read_tx(B#block.txs), not is_record(TX, tx)]` was misleading here.